### ISSUE TITLE
String escapes

### ISSUE THREAD

**Issue by laurentlb** [bazelbuild/starlark#38]
The specification doesn't describe string literals.
I tried this code. Each implementation of Starlark has a different result.

```
print("\a")
print("\b")
print("\c")
...
print("\z")
```

TODO:
 * Agree on the escapes we handle.
 * What shall we do with unknown escapes (e.g. `\z`). All implementations and Python return two characters `\` and `z`. This behavior will prevent us to support more escapes in the future (as it would break users).

The fact that the `\` character is sometimes used as an escape, and sometimes as a literal character is confusing.

Example of bug found in user code:
```python
s = s.replace("'", "\'").replace('"', '\"')
```

**Comment by laurentlb** [comment]
From the [Python manual](https://docs.python.org/3/reference/lexical_analysis.html#literals):

> Changed in version 3.6: Unrecognized escape sequences produce a DeprecationWarning. In some future version of Python they will be a SyntaxError.

I'd propose to make this change in Starlark too.

**Comment by laurentlb** [comment]
cc @Quarz0 @alandonovan
We'd like to make this change in Bazel soon.

**Comment by alandonovan** [comment]
I agree that rejecting unrecognized escape sequences (like C/C++/Go/Java) is the right approach. 
Migration should not be hard because the problem can be reliably identified during lexing.
I don't think we should add deprecation warnings: 99.9% of people forced to read them are powerless to do anything to address the problem.

**Comment by laurentlb** [comment]
Agree. My comment was unclear, I wanted to suggest only the syntax error (Bazel has a way to roll out incompatible changes without using warnings).

**Comment by brandjon** [comment]
This appears to be implemented in both the Java and Go versions and is also reflected in the spec. Closing.
